File: packages/python/src/output_guard/formats/json_.py
"""json format: parse JSON, falling back to json-repair on syntax error.

Mirror of packages/ts/src/formats/json.ts.
"""

from __future__ import annotations

import json
from typing import Any

from json_repair import repair_json as _json_repair

from ..errors import ParseError


def parse_json(text: str) -> Any:
    try:
        return json.loads(text)
    except json.JSONDecodeError:
        try:
            repaired = _json_repair(text)
            return json.loads(repaired)
        except (json.JSONDecodeError, ValueError) as e:
            raise ParseError(f"JSON parse failed: {e}", e) from e


def repair_json(text: str) -> str:
    try:
        return _json_repair(text)
    except (ValueError, TypeError) as e:
        raise ParseError(f"json-repair failed: {e}", e) from e

File: packages/ts/src/formats/json.ts
/**
 * json format: parse JSON, falling back to jsonrepair on syntax error.
 */
import { jsonrepair } from "jsonrepair";
import { ParseError } from "../errors.js";

export const parseJson = (text: string): unknown => {
  try {
    return JSON.parse(text);
  } catch {
    // fallback: try jsonrepair
    try {
      const repaired = jsonrepair(text);
      return JSON.parse(repaired);
    } catch (e) {
      throw new ParseError(`JSON parse failed: ${String(e)}`, e);
    }
  }
};

export const repairJson = (text: string): string => {
  try {
    return jsonrepair(text);
  } catch (e) {
    throw new ParseError(`jsonrepair failed: ${String(e)}`, e);
  }
};
